lib/internal/Magento/Framework/Module/ModuleList/Loader.php

```diff
@@ -48,24 +48,32 @@ class Loader
     private $filesystemDriver;
 
     /**
+     * @var Sorter
+     */
+    private $sorter;
+
+    /**
      * Constructor
      *
      * @param Dom $converter
      * @param Parser $parser
      * @param ComponentRegistrarInterface $moduleRegistry
      * @param DriverInterface $filesystemDriver
+     * @param Sorter $sorter
      */
     public function __construct(
         Dom $converter,
         Parser $parser,
         ComponentRegistrarInterface $moduleRegistry,
-        DriverInterface $filesystemDriver
+        DriverInterface $filesystemDriver,
+        Sorter $sorter
```

Comment:
In this case I think the `Sorter` parameter should be optional http://devdocs.magento.com/guides/v2.2/contributor-guide/backward-compatible-development/#adding-a-constructor-parameter